LoopFollow/ViewControllers/MealViewController.swift

```diff
@@ -91,42 +91,61 @@ class MealViewController: UIViewController, UITextFieldDelegate {
         }
         
         // CARBS & FPU ENTRIES
-        // Convert carbGrams, fatGrams, and proteinGrams to integers or default to 0 if empty
-        let carbs: Int
-        let fats: Int
-        let proteins: Int
+        guard var carbText = carbGrams.text else {
+            print("Error: Carb amount not entered")
+            return
+        }
         
-        if let carbText = carbGrams.text, !carbText.isEmpty {
-            guard let carbsValue = Int(carbText) else {
-                print("Error: Carb input value conversion failed")
+        carbText = carbText.replacingOccurrences(of: ",", with: ".")
+        
+        let carbsValue: Double
+        if carbText.isEmpty {
+            carbsValue = 0
+        } else {
+            guard let carbsDouble = Double(carbText) else {
+                print("Error: Carb amount conversion failed")
                 return
             }
-            carbs = carbsValue
-        } else {
-            carbs = 0
+            carbsValue = carbsDouble
+        }
+        
+        guard var fatText = fatGrams.text else {
+            print("Error: Fat amount not entered")
+            return
         }
         
-        if let fatText = fatGrams.text, !fatText.isEmpty {
-            guard let fatsValue = Int(fatText) else {
-                print("Error: Fat input value conversion failed")
+        fatText = fatText.replacingOccurrences(of: ",", with: ".")
+        
+        let fatsValue: Double
+        if fatText.isEmpty {
+            fatsValue = 0
+        } else {
+            guard let fatsDouble = Double(fatText) else {
+                print("Error: Fat amount conversion failed")
                 return
             }
-            fats = fatsValue
-        } else {
-            fats = 0
+            fatsValue = fatsDouble
         }
         
-        if let proteinText = proteinGrams.text, !proteinText.isEmpty {
-            guard let proteinsValue = Int(proteinText) else {
-                print("Error: Protein input value conversion failed")
+        guard var proteinText = proteinGrams.text else {
+            print("Error: Protein amount not entered")
+            return
+        }
+        
+        proteinText = proteinText.replacingOccurrences(of: ",", with: ".")
+        
+        let proteinsValue: Double
+        if proteinText.isEmpty {
+            proteinsValue = 0
+        } else {
+            guard let proteinsDouble = Double(proteinText) else {
+                print("Error: Protein amount conversion failed")
                 return
             }
-            proteins = proteinsValue
-        } else {
-            proteins = 0
+            proteinsValue = proteinsDouble
         }
         
-        if carbs > maxCarbs || fats > maxCarbs || proteins > maxCarbs {
+        if carbsValue > maxCarbs || fatsValue > maxCarbs || proteinsValue > maxCarbs {
```

Comment:
Comparing fat and protein to max carb threshold seems odd. Is that intentional?

Reply:
Intentional (and been there since introducing maxcarbs), but as discussed in the "disable send buttons"-issue, this needs more thought. Will post an update to this PR shortly and comment out the protein/fat guardrails for now until we agreed on a better implementation.

Reply:
Commented out

```swift
|| fats > maxCarbs || proteins > maxCarbs
``` 

Check if it looks ok and then merge this PR (I fix the potential merge conflict in the other PR after that (its the infodisplaysettingview force unwrap crash code that I had to update in both feature branches)